rust/azure_iot_operations_mqtt/src/connection_settings.rs

```diff
@@ -4,6 +4,7 @@
 //! Generic MQTT connection settings implementations
 
 use std::env::{self, VarError};
+use std::path::{Path, PathBuf};
```

Comment:
why path buf

Reply:
`PathBuf` can be a source of truth that we can use slice references to (Both `Path`s and `str`s) that reduce the number of allocations necessary. Before, all the `String` concatenations and `format!`s were requiring allocations.